lib/puppet/pops/issues.rb

```diff
@@ -58,7 +58,8 @@ def format(hash ={})
   # in the issue's message producing block.
   # @api private
   #
-  class MessageData
+  class MessageData < Puppet::Pops::LabelProvider
```

Comment:
You can do a method missing, and delegate to the provider if it is set, and it it responds to the missing method.